command/node_status.go

```diff
@@ -231,24 +231,14 @@ func (c *NodeStatusCommand) Run(args []string) int {
 			out[0] += "|Running Allocs"
 		}
 
-		queryOptions := &api.QueryOptions{AllowStale: true}
-		var nodeInfo *api.Node
-
 		for i, node := range nodes {
-			if c.os {
-				nodeInfo, _, err = client.Nodes().Info(node.ID, queryOptions)
-				if err != nil {
-					c.Ui.Error(fmt.Sprintf("Error getting node info: %s", err))
-					return 1
-				}
-			}
 			out[i+1] = fmt.Sprintf("%s|%s|%s|%s",
 				limit(node.ID, c.length),
 				node.Datacenter,
 				node.Name,
 				node.NodeClass)
 			if c.os {
-				out[i+1] += fmt.Sprintf("|%s", nodeInfo.Attributes["os.name"])
+				out[i+1] += fmt.Sprintf("|%s", node.Attributes["os.name"])
```

Comment:
Do you want `os.name` (eg `ubuntu`, `centos`) or `kernel.name` (eg `linux`, `windows`)?

I was thinking displaying `fmt.Sprintf("%s/%s", attrs["kernel.name"], attrs["cpu.arch"])` would give you really useful output for heterogeneous clusters: `linux/amd64`, `linux/arm64`, etc.

Reply:
@schmichael My initial intention was only `os.name`  (`ubuntu`, `centos`). However after reading your comment, I would love to have both included with `-os` flag. e.g. how does this look?

```
$ nomad node status -os
ID        DC   Name   Class   OS      Drain  Eligibility  Status Arch
789cdb6d  dc1  linux  <none>  ubuntu  false  eligible     ready linux/amd64
```

Reply:
Let me know if this is okay, and I will update the PR.

Reply:
Do you need the distro name as well? I was kind of hoping Arch (`linux/amd64`, `windows/amd64`, etc) would be sufficient for most use cases.

Reply:
Distro name is really what we need 🙂 . e.g. In our environment `centos` is running some old version of Nomad and I would like to see which nomad clients are still running on `centos` so we can identify and upgrade them. This is just one example at the top of my head.

For me `Arch` is optional but nice to have!

Reply:
@schmichael Any updates on this one? I think we should address this soon since the current one doesn't work 🙂 . 
See my comment [`here`](https://github.com/hashicorp/nomad/pull/12388#issuecomment-1089363767)

Reply:
Sorry we can ship this as-is. `Arch` can always be added later.

Reply:
Sounds good! Let me know if there are any docs that needs to be updated.

Reply:
Sorry! LGTM to me. Since it's behind the `-os` flag that buys us a lot of leeway.

Reply:
Good call! API docs will need updating https://www.nomadproject.io/api-docs/nodes#parameters

Reply:
Fixed.